### PR TITLE
merge restructed shop to development

### PR DESCRIPTION
**Pull request by hskim881028**
### Description
SSIA

### Required Reviewers
* @planetarium/9c-dev

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1141562434100787/1200534677193093) by [Unito](https://www.unito.io)

### CODE REVIEW

File: nekoyume/Assets/_Scripts/UI/Model/ShopItem.cs [hunk 1]
```diff
@@ -10,59 +10,34 @@ namespace Nekoyume.UI.Model
...
+            Level.Value = orderDigest.Level;
         }
 
         public override void Dispose()
         {
-            SellerAgentAddress.Dispose();
-            SellerAvatarAddress.Dispose();
             Price.Dispose();
-            ProductId.Dispose();
+            OrderId.Dispose();
```
Comment (boscohyun) [inline nekoyume/Assets/_Scripts/UI/Model/ShopItem.cs:39]:
```suggestion
            OrderId.Dispose();
            TradableId.Dispose();
            ExpiredBlockIndex.Dispose();
            Level.Dispose();
```

File: nekoyume/Assets/_Scripts/UI/Model/ShopItem.cs [hunk 2]
```diff
@@ -10,59 +10,34 @@ namespace Nekoyume.UI.Model
...
-                item.TradableFungibleItemCount, GetItemBase(item), item.ExpiredBlockIndex)
+        public ShopItem(OrderDigest orderDigest) : this(orderDigest, Util.CreateItemBaseByItemId(orderDigest.ItemId))
         {
-
         }
 
-        private ShopItem(Address sellerAgentAddress, Address sellerAvatarAddress,
-                         FungibleAssetValue price, Guid productId, int count,
-                         ItemBase item, long expiredBlockIndex) : base(item, 1)
+        private ShopItem(OrderDigest orderDigest, ItemBase item) : base(item, 1)
```
Comment (boscohyun) [inline nekoyume/Assets/_Scripts/UI/Model/ShopItem.cs:25]:
```suggestion
        private ShopItem(OrderDigest orderDigest, ItemBase item) : base(item, orderDigest.ItemCount)
```

File: nekoyume/Assets/_Scripts/UI/Model/ShopItem.cs [hunk 3]
```diff
@@ -10,59 +10,34 @@ namespace Nekoyume.UI.Model
...
             GradeEnabled.Value = true;
-            SellerAgentAddress.Value = sellerAgentAddress;
-            SellerAvatarAddress.Value = sellerAvatarAddress;
-            Price.Value = price;
-            Count.Value = count;
-            ProductId.Value = productId;
-            ItemSubType.Value = item.ItemSubType;
-            ExpiredBlockIndex.Value = expiredBlockIndex;
+            Price.Value = orderDigest.Price;
+            Count.Value = orderDigest.ItemCount;
```
Comment (boscohyun) [inline nekoyume/Assets/_Scripts/UI/Model/ShopItem.cs:29]:
```suggestion
```

File: nekoyume/Assets/_Scripts/UI/Model/ShopItems.cs [hunk 4]
```diff
@@ -108,220 +66,120 @@ public void Dispose()
...
-            isMultiplePurchase = value;
+            return item;
         }
 
-        public void SelectItemView(ShopItemView view)
+        protected void SelectItemView(ShopItemView view)
         {
-            if (view is null ||
-                view.Model is null)
+            if (view?.Model is null)
```
Comment (boscohyun) [inline nekoyume/Assets/_Scripts/UI/Model/ShopItems.cs:87]: Is there no bypass warns in `view?.Model`?

File: nekoyume/Assets/_Scripts/UI/Model/ShopItems.cs [hunk 5]
```diff
@@ -108,220 +66,120 @@ public void Dispose()
...
-                {
-                    continue;
-                }
-
-                target.Dispose();
-                pair.Value.Remove(target);
-            }
-        }
-
+        // public void RemoveAgentProduct(Guid productId)
```
Comment (boscohyun) [inline nekoyume/Assets/_Scripts/UI/Model/ShopItems.cs:115]: These comments seems useless.

File: nekoyume/Assets/_Scripts/UI/Shop/ShopBuy.cs [hunk 6]
```diff
@@ -116,16 +103,10 @@ private async void AsyncShow(bool ignoreShowAnimation = false)
...
-                    BottomMenu.ToggleableType.Mail,
-                    BottomMenu.ToggleableType.Quest,
-                    BottomMenu.ToggleableType.Chat,
-                    BottomMenu.ToggleableType.IllustratedBook,
-                    BottomMenu.ToggleableType.Ranking,
-                    BottomMenu.ToggleableType.Character);
+                Find<BottomMenu>().Show(UINavigator.NavigationType.Back, SubscribeBackButtonClick,
+                    true, BottomMenu.ToggleableType.Mail, BottomMenu.ToggleableType.Quest,
+                    BottomMenu.ToggleableType.Chat, BottomMenu.ToggleableType.IllustratedBook,
+                    BottomMenu.ToggleableType.Ranking, BottomMenu.ToggleableType.Character);
```
Comment (boscohyun) [inline nekoyume/Assets/_Scripts/UI/Shop/ShopBuy.cs:109]: Is this intended? 😥

**Comment by boscohyun** [comment]
There is some conflicts. 😥

File: nekoyume/Assets/_Scripts/BlockChain/ActionRenderHandler.cs [hunk 7]
```diff
@@ -286,20 +289,25 @@ private void ResponseRapidCombination(ActionBase.ActionEvaluation<RapidCombinati
                 //[TentuPlay] RapidCombinationConsumable 합성에 사용한 골드 기록
                 //Local에서 변경하는 States.Instance 보다는 블락에서 꺼내온 eval.OutputStates를 사용
```
Comment (Unengine) [inline nekoyume/Assets/_Scripts/BlockChain/ActionRenderHandler.cs:290]: TentuPlay should be removed.

**Review by boscohyun — APPROVED** [review]

**Review by Unengine — APPROVED** [review]

**Pull request merged**